credentials/tls.go

```diff
@@ -110,14 +110,14 @@ func (c tlsCreds) Info() ProtocolInfo {
 func (c *tlsCreds) ClientHandshake(ctx context.Context, authority string, rawConn net.Conn) (_ net.Conn, _ AuthInfo, err error) {
 	// use local cfg to avoid clobbering ServerName if using multiple endpoints
 	cfg := credinternal.CloneTLSConfig(c.config)
-	if cfg.ServerName == "" {
-		serverName, _, err := net.SplitHostPort(authority)
-		if err != nil {
-			// If the authority had no host port or if the authority cannot be parsed, use it as-is.
-			serverName = authority
-		}
-		cfg.ServerName = serverName
+
```

Comment:
Should we document on the `NewTLS` function that `config.ServerName` is ignored?

Reply:
It's not ignored, actually.  It's used by the channel in the logic mentioned in the first PR comment, as long as the user didn't manually override the channel's authority.

https://github.com/grpc/grpc-go/blob/9fa3267859958a7fa0141a8180102850f3d5842c/clientconn.go#L1810-L1813

https://github.com/grpc/grpc-go/blob/9fa3267859958a7fa0141a8180102850f3d5842c/credentials/tls.go#L106